c2corg_ui/static/js/style/progressbar.js

```diff
@@ -0,0 +1,191 @@
+ */
+app.progressBarDirective = function() {
+  return {
+    restrict: 'EA',
+    controller: 'appProgressBarController as progBarCtrl',
+    bindToController: true
+  };
+};
+
+app.module.directive('appProgressBar', app.progressBarDirective);
+
+
+/**
+ * @constructor
+ * @export
+ */
+app.ProgressBarController = function() {
+
+  /**
+   * first next step would be 2 by default
+   * @export
+   */
+  this.next_step = 2;
+
+  /**
+   * Current step is 1 by default
+   * @export
+   */
+  this.current_step = 1;
+
+  /**
+   * Previous step is 0 by default
+   * @export
+   */
+  this.previous_step = 0;
+
+  /**
+   * Max possible steps for creation/edition
+   * @export
+   */
+  this.max_steps;
+
+  /**
+   * Waypoint type initialised in the edit.html
+   * @export
+   */
+  this.waypoint_type;
+};
+
+
+/**
+ * Navigate through creation/editing steps
+ * @param {number} step
+ * @param {string} direction
+ * @export
+ */
+app.ProgressBarController.prototype.step = function(step, document, direction) {
+  switch (step) {
+    case 1:
+      $('.editing').animate({left: '0'});
+      this.animateBar_(step, direction);
+      this.previous_step = 0;
+      this.current_step = 1;
+      this.next_step = 2;
+      break;
+
+    case 2:
+      $('.editing').animate({left: '-115%'});
+      this.animateBar_(step, direction);
+      this.previous_step = 1;
+      this.current_step = 2;
+      this.next_step = 3;
+      break;
+
+    case 3:
+      $('.editing').animate({left: '-229%'});
+      this.animateBar_(step, direction);
+      this.previous_step = 2;
+      this.current_step = 3;
+      this.next_step = 4;
+      break;
+
+    case 4:
+      $('.editing').animate({left: '-343%'});
+      this.animateBar_(step, direction);
+      this.previous_step = 3;
+      this.current_step = 4;
+      this.next_step = 5;
+      break;
+
+    case 5:
+      $('.editing').animate({left: '-457%'});
+      this.animateBar_(step, direction);
+      this.previous_step = 4;
+      this.current_step = 5;
+      this.next_step = 6;
+      break;
+
+    default:
+      break;
+  }
+};
+
+
+/**
+ * Animate the progress bar
+ * @param {number} step
+ * @param {string} direction
+ * @private
+ */
+app.ProgressBarController.prototype.animateBar_ = function(step, direction) {
+  var percent = 100 / this.max_steps;
+  var green = '#7EFF1F'; // completed color
+  var gray = '#B4B4B4'; // left color
+  var willBe;
+  var nextPosition;
+  var stopBack;
+
+  $('.nav-step-selected').removeClass('nav-step-selected');
+  $('.nav-step-' + step).addClass('nav-step-selected');
+  $('html, body').animate({scrollTop: 0}, 'slow');
+
+  if (direction === 'forwards') {
+    willBe = (percent * (step - 1)) - 10;
+    nextPosition = (percent * step) - 10;
+  } else {
+    willBe = (percent * step) - 10;
+    nextPosition = (percent * (step + 1)) - 10;
+    stopBack = willBe;
+  }
+
+  // bar animation, timeout for a gradual filling
+  var timeout = setInterval(function() {
```

Comment:
If there is any advantage of using Angular instead of the normal `setInterval` that has worked just well until now, maybe yes?